### PR TITLE
feat(confirmation-modal): ✨ Add confirmation modal with relevant hooks

### PR DESCRIPTION
**Pull request by tukantje**
# Summary
When our application is loaded in an iframe, we have UX breaking issues whenever we'd like to confirm with the user if they'd like to take certain actions, such as removing a list.

This happens since we are relying on `window.prompt` which is a sandboxed feature. So in places such as Safe app we end up with a broken experience.

This PR adds a new feature: `useConfirmationRequest`. This hook can be used anywhere, to request for the user to confirm whether or not they'd like to confirm what they are doing.

As an example, I've refactored the **Remove List** check which normally looks as following;

Screenshot 2023-04-24 at 17 56 28

After these changes, it functions as the following;

Screenshot 2023-04-24 at 17 53 21

# To Test

1. Go to application
2. Click on a currency
3. Click on manage token lists
4. Try to remove one via the gear icon, and click on remove list
5. You should see the new confirmation modal

# Background
https://cowservices.slack.com/archives/C0361CDG8GP/p1682078523149949

# Follow up
Once approach is agreed upon, clearing out all the `window.prompt` calls which should be easy.

### CODE REVIEW

**Review by shoom3301 — APPROVED** [review]
It's perfect! Everything looks awesome!
Only one think I would wish - can you replace `confirmPriceImpactWithoutFee` function by the new solution, please?

**Comment by tukantje** [comment]
>  Only one think I would wish - can you replace confirmPriceImpactWithoutFee function by the new solution, please?

On it 🫡

**Review by nenadV91 — APPROVED** [review]
Codewise LGTM 👍

**Review by elena-zh — APPROVED** [review]
Great job!

**Review by shoom3301 — APPROVED** [review]
Awesome!

**Comment by elena-zh** [comment]
Hey @tukantje , price impact modal looks awesome. 
However, there is 1 case that is not covered by the logic: when a price impact >5% and less than 10%, we don't make  a user type in the 'confirm' work. We just make him press on the OK button. 
See the modal:
ok

In the current implementation the modal is:
press ok

Could you please fix it?

Tnx

**Comment by tukantje** [comment]
@elena-zh good point. Fixed it, it is deploying.

**Comment by elena-zh** [comment]
Looks great to me now!

**Review by anxolin — COMMENTED** [review]
👍 ony nit, is that our common module is growing a bit. I think in the future we should create more modules and organise code better

File: src/components/SearchModal/ManageLists/ManageListsMod.tsx [hunk 1]
```diff
@@ -131,13 +139,16 @@ const ListRow = memo(function ListRow({
...
-
+    removeListAnalyticsWithConfirm({
+      title: t`Remove this list?`,
+      description: t`This action will remove from your lists: ${list?.name ?? 'unknown'}`,
+      confirmWord: t`remove`,
+      callToAction: t`Remove list`,
+      warning: t`Are you sure? You will need to add this list by its url, if you would like to restore it later on.`,
+      action: t`remove list`,
+    })
+    // }, [dispatch, listUrl])
```
Comment (alfetopito) [inline src/components/SearchModal/ManageLists/ManageListsMod.tsx:150]: Dead code?

**Pull request merged**